### PR TITLE
Use cURL instead of wget.

### PR DESCRIPTION
**Pull request by wmark**
This enables us to use schemes like scp://, smb:// and file://.
The latter, when used in BASE_URL, obsoletes option "-t TMPDIR".

cURL is already used in another script; this fixes the mishmash.

### CODE REVIEW

File: bin/coreos-install [hunk 1]
```diff
@@ -224,15 +224,17 @@ if [[ -z "${BASE_URL}" ]]; then
     BASE_URL="http://${CHANNEL_ID}.release.core-os.net/amd64-usr"
 fi
 IMAGE_URL="${BASE_URL}/${VERSION_ID}/${IMAGE_NAME}"
+: ${IMAGE_HTTP_HEADER:="Accept: application/x-bzip, application/octet-stream"}
+: ${SIG_HTTP_HEADER:="Accept: application/pgp-signature"}
 SIG_NAME="${IMAGE_NAME}.sig"
 SIG_URL="${BASE_URL}/${VERSION_ID}/${SIG_NAME}"
 
-if ! wget --inet4-only --spider --quiet "${IMAGE_URL}"; then
+if ! curl -4 --head -sS --fail -H "${IMAGE_HTTP_HEADER}" -LR "${IMAGE_URL}"; then
```
Comment (crawford) [inline bin/coreos-install]: Can you use long flags for all of these?

**Comment by marineam** [comment]
Hm, I would say this is fine except for it was previously changed from curl to wget in order to support using this script in some minimal environments. wget is a little more common than curl. See 5a7c672e
Of course I don't really know if any of the other changes made since then continue to work whatever environment inspired the switch but that's at least the reason for why it is the way it is.

Thoughts? a wrapper function could be used to select which ever is available but I'm inclined to just defer polish like seamlessly supporting other protocols for when this script finally gets replaced by a more comprehensive installer.

**Comment by wmark** [comment]
Yes, I did notice that change, but nevertheless guess that there are more users who benefit from this change than those who are stuck with *wget*.

As compromise we could add checks if an image file already exists and skip downloading, and by this cURL and wget, althogether.

**Comment by marineam** [comment]
Yeah, if the base url happens to be a directory path that exists skipping the download bits would be reasonable. That's a bit more friendly than having to use the file:// scheme too. To avoid further complicating the streaming download+verify+write magic perhaps just wrap all the existing code in an if or function and write a new bit that just does a simple verify and write.

**Comment by wmark** [comment]
Implementing your suggestions. Testing will take some time.

**Comment by wmark** [comment]
It's been either duplicating ```WORKDIR=…; gpg --import``` or ```if [[ -d "${BASE_URL#file://}" ]]; then```. The latter guards us at least from a gone-away directory.

Everything else (except the pre-flight check) is just an intendeted copy of the last and previous version of *coreos-install*.

**Comment by marineam** [comment]
Looks reasonable, any reason for still switching from wget to curl at this point? Part of the reason for this logic was to keep wget around.

**Comment by wmark** [comment]
Still the aforementioned and the commented. Currently there *wget* here and *cURL* in other scripts.

Either we will make *wget* or *cURL* users sad. cURL offers more.

**Comment by marineam** [comment]
This script is special in that it is used as a stand-alone tool while other scripts in this repo are only needed in CoreOS. It's placement here is kinda arbitrary. Lets stick with wget for now.

**Pull request closed without merging**